src/edit/PosterFrameUpload.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <tailor-dialog
+    v-model="dialog"
+    @click:outside="reset"
+    header-icon="mdi-image-multiple"
+    width="608">
+    <template #activator="{ on, attrs }">
+      <v-btn
+        v-on="on"
+        v-bind="attrs"
+        :disabled="isDisabled"
+        text>
+        <v-icon class="mr-2">mdi-image-multiple</v-icon>
+        Select poster frame
+      </v-btn>
+    </template>
+    <template #header>
+      Poster frame
+    </template>
+    <template #body>
+      <p class="mb-3 text-left">
+        Set an image that's displayed before the video is played.
+      </p>
+      <v-img :src="currentPosterFrame" class="frame" />
+      <p class="mt-7 mb-3 text-left">
+        Pick a frame from the video
+      </p>
+      <div class="poster-frames-container">
+        <poster-frame
+          v-for="(poster, index) in generatedPosterFrames"
+          :key="poster"
+          @click="selectedIndex = index"
+          :src="poster"
+          :is-selected="!image && (selectedIndex === index)" />
+      </div>
+      <p class="mt-3 mb-0 text-left">
+        or upload an image from your computer
+      </p>
+      <div class="mt-3 text-left">
+        <upload-btn
+          @change="upload"
+          label="Upload image"
+          accept="image/jpeg"
+          small depressed>
+          <template #icon>
+            <v-icon>mdi-upload</v-icon>
+          </template>
+        </upload-btn>
+        <p :class="{ 'error--text': isError }" class="mt-1 text-caption">
+          Poster frame must be under 500KB.
+        </p>
+      </div>
+    </template>
+    <template #actions>
+      <v-btn @click="reset" color="primary" text>Close</v-btn>
+      <v-btn @click="save" color="success" text>Save</v-btn>
+    </template>
+  </tailor-dialog>
+</template>
+
+<script>
+import isNil from 'lodash/isNil';
+import PosterFrame from './PosterFrame.vue';
+import TailorDialog from '../tce-core/TailorDialog.vue';
+import take from 'lodash/take';
+import UploadBtn from './UploadBtn.vue';
+
+const MAX_SIZE = 500000; // 500 KB
+const CUSTOM_POSTER_FRAME_INDEX = 4;
+
+export default {
+  name: 'poster-frame-upload',
+  props: {
+    id: { type: String, default: null },
+    playable: { type: Boolean, default: false },
+    selectedPosterFrameIndex: { type: Number, default: 0 },
+    posterFrames: { type: Array, default: () => ([]) }
+  },
+  data: () => ({
+    dialog: false,
+    selectedIndex: null,
```

Comment:
Can we derive state from props to have a single source of truth within the component?
```suggestion
    selectedIndex: this.selectedPosterFrameIndex,
```